Description: Changes Haskell generation to not use unqualified modules
 Recent base releases are incompatible with haskell98, which provides several
 unqualified modules.  Here we change Haskell generation to generate
 qualified modules, so that haskell98 is not required to compile the
 generated code.
Author: Antti-Juhani Kaijanaho <ajk@debian.org>
Forwarded: by email to Markus Forsberg <markus.forsberg@gu.se> and Aarne Ranta <aarne@chalmers.se>
Last-Update: 2012-04-01
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/formats/haskell-gadt/HaskellTopGADT.hs
+++ b/formats/haskell-gadt/HaskellTopGADT.hs
@@ -231,8 +231,8 @@
	        ["-- automatically generated by BNF Converter",
		 "module Main where\n",
	         "",
-	         "import IO ( stdin, hGetContents )",
-	         "import System ( getArgs, getProgName )",
+	         "import System.IO ( stdin, hGetContents )",
+	         "import System.Environment ( getArgs, getProgName )",
		 "",
		 "import " ++ alexFileM     opts,
		 "import " ++ happyFileM    opts,
--- a/formats/haskell2/HaskellTop.hs
+++ b/formats/haskell2/HaskellTop.hs
@@ -241,8 +241,8 @@
	        ["-- automatically generated by BNF Converter",
		 "module Main where\n",
	         "",
-	         "import IO ( stdin, hGetContents )",
-	         "import System ( getArgs, getProgName )",
+	         "import System.IO ( stdin, hGetContents )",
+	         "import System.Environment ( getArgs, getProgName )",
		 "",
		 "import " ++ alexFileM     opts,
		 "import " ++ happyFileM    opts,
--- a/formats/profile/ProfileTop.hs
+++ b/formats/profile/ProfileTop.hs
@@ -184,8 +184,8 @@
	         "",
                  "import Trees",
                  "import Profile",
-	         "import IO ( stdin, hGetContents )",
-	         "import System ( getArgs, getProgName )",
+	         "import System.IO ( stdin, hGetContents )",
+	         "import System.Environment ( getArgs, getProgName )",
		 "",
		 "import " ++ alexFileM     inDir name,
		 "import " ++ happyFileM    inDir name,
